### PR TITLE
Add TrustyAI operator to ODH

### PR DESCRIPTION
**Pull request by RobGeada**
## Description

- Adds TrustyAI Operator to manifests
- Removes TrustyAI-Service from manifests
- Changes TrustyAI tests from service-based to operator-based

## How Has This Been Tested?

- Extensively tested via TrustyAI's CI; the test scripts here are modifications of our upstream CI
- Tested via the UI team during their development of the TrustyAI dashboard

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: tests/resources/trustyai/trustyai_crd.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: trustyai.opendatahub.io.trustyai.opendatahub.io/v1alpha1
+kind: TrustyAIService
+metadata:
+  name: trustyai-service
+spec:
+  # Optional values for replicas, image and tag. Below are the default values.
```
Comment (ruivieira) [inline tests/resources/trustyai/trustyai_crd.yaml:6]: I think these comments can be deleted, as they are documented elsewhere. wdyt?

File: trustyai-service-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+Please ensure the namespace specified is the same as the namespace where you have deployed the operator.
+
+After the ConfigMap is applied, you can then proceed to deploy the TrustyAI service.
+The operator will use the image name and tag specified in the ConfigMap for the deployment.
+
+If you want to use a different image or tag in the future, you'll need to update the ConfigMap and redeploy the operator to have the changes take effect. The running TrustyAI services won't be redeployed automatically. To use the new image or tag, you'll need to delete and recreate the TrustyAIService resources.
+
+## Contributing
```
Comment (ruivieira) [inline trustyai-service-operator/README.md:135]: These two links will be broken. I would link to upstream `trustyai-service-operator`s repos files.

File: trustyai-service-operator/manager/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+resources:
+- manager.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+images:
+- name: controller
+  newName: quay.io/trustyai/trustyai-service-operator
+  newTag: latest
```
Comment (ruivieira) [inline trustyai-service-operator/manager/kustomization.yaml:8]: We should probably change this tag to a fixed one.

**Review by ruivieira — CHANGES_REQUESTED** [review]

File: tests/resources/trustyai/trustyai_crd.yaml (hunk 1 above)
Comment (RobGeada) [reply tests/resources/trustyai/trustyai_crd.yaml:6]: done

File: trustyai-service-operator/README.md (hunk 2 above)
Comment (RobGeada) [reply trustyai-service-operator/README.md:135]: Done

File: trustyai-service-operator/manager/kustomization.yaml (hunk 3 above)
Comment (RobGeada) [reply trustyai-service-operator/manager/kustomization.yaml:8]: done

**Review by ruivieira — APPROVED** [review]

**Comment by LaVLaS** [comment]
/retest

@RobGeada Can you add an OWNERS file so that the tide plugin knows who to ping for reviews.

Also, the openshift-ci system is still experiencing issues so we'll auto-merge this once the openshfit-ci issues are resolved

**Comment by RobGeada** [comment]
> /retest
> 
> @RobGeada Can you add an OWNERS file so that the tide plugin knows who to ping for reviews.
> 
> Also, the openshift-ci system is still experiencing issues so we'll auto-merge this once the openshfit-ci issues are resolved

@LaVLaS I still have a few items to attend to on this PR beforehand, can we mark it as a `do-not-merge` in some way for now?

**Comment by RobGeada** [comment]
> /retest
> 
> @RobGeada Can you add an OWNERS file so that the tide plugin knows who to ping for reviews.
> 
> Also, the openshift-ci system is still experiencing issues so we'll auto-merge this once the openshfit-ci issues are resolved

The OWNERS file is [here](https://github.com/opendatahub-io/odh-manifests/blob/a4721a876986d3bdea593c09c2b1cd6bef2c0b16/trustyai-service-operator/OWNERS)

**Review by danielezonca — APPROVED** [review]

**Comment by LaVLaS** [comment]
The openshift-ci issues may still be affecting ocp 4.12 test runs.  If the OCP 4.11 test run succeeds without any errors in our CI test then I'll admin override the merge

**Comment by LaVLaS** [comment]
OCP 4.12 failed with the same known issue that is blocking the installation of any OLM operators on the cluster.  
Since the OCP4.11 test ran without issue and the OCP 4.12 issue is outside of our control, I am merging this.  The OCP 4.12 failure should be resolved later this week

**Pull request merged**